### PR TITLE
[crypto] Implement ECDSA and ECDH using PSA crypto API

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
#### Problem
Moving crypto operation implementation to PSA crypto API.

#### Change overview
The ECDSA and ECDH operations specified by P256Keypair and P256PublicKey classes have been implemented using PSA crypto API provided by mbedTLS 3.1.0.

#### Testing
Tested by running `out/ut/tests/CHIPCryptoPALTest` and checking memory leaking using Valgrind.

### CODE REVIEW

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 1]
```diff
@@ -739,113 +706,113 @@ bool IsBufferContentEqualConstantTime(const void * a, const void * b, size_t n)
...
     VerifyOrExit(bbuf.Fit(), error = CHIP_ERROR_BUFFER_TOO_SMALL);
 
     output.SetLength(bbuf.Needed());
 
 exit:
-    ClearSecretData(privkey, sizeof(privkey));
-    logMbedTLSError(result);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Crypto, "P256Keypair serialization error %s", error.AsString());
```
Comment (Damian-Nordic) [inline src/crypto/CHIPCryptoPALPSA.cpp:779]:
```suggestion
        ChipLogError(Crypto, "P256Keypair serialization error " CHIP_LOG_FORMAT, error.Format());
```
This produces smaller code when verbose error messages are disabled.

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 2]
```diff
@@ -521,189 +518,159 @@ mbedtls_ecp_group_id MapECPGroupId(SupportedECPKeyTypes keyType)
...
-    VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
+    status = psa_import_key(&attributes, keypair->mPrivateKey, PSA_BITS_TO_BYTES(keypair->mBitLength), &keyId);
+    VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
 
-    VerifyOrExit(out_signature.SetLength(kP256_ECDSA_Signature_Length_Raw) == CHIP_NO_ERROR, error = CHIP_ERROR_INTERNAL);
+    status = psa_sign_message(keyId, PSA_ALG_ECDSA(PSA_ALG_SHA_256), msg, msg_length, out_signature.Bytes(),
+                              out_signature.Capacity(), &outputLen);
+    VerifyOrExit(status == PSA_SUCCESS, error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(outputLen == kP256_ECDSA_Signature_Length_Raw, error = CHIP_ERROR_INTERNAL);
+    SuccessOrExit(out_signature.SetLength(outputLen) == CHIP_NO_ERROR);
```
Comment (kkasperczyk-no) [inline src/crypto/CHIPCryptoPALPSA.cpp:561]: We will go to the `exit` label after this check anyway and now we in fact don't assign any information to the error variable. How about just replacing it with `error = out_signature.SetLength(outputLen);`?

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 3]
```diff
@@ -521,189 +518,159 @@ mbedtls_ecp_group_id MapECPGroupId(SupportedECPKeyTypes keyType)
...
-#endif
 }
 
 CHIP_ERROR P256PublicKey::ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length,
                                                        const P256ECDSASignature & signature) const
 {
-#if defined(MBEDTLS_ECDSA_C)
-    VerifyOrReturnError((msg != nullptr) && (msg_length > 0), CHIP_ERROR_INVALID_ARGUMENT);
+    CHIP_ERROR error                = CHIP_NO_ERROR;
+    psa_status_t status             = PSA_ERROR_BAD_STATE;
```
Comment (kkasperczyk-no) [inline src/crypto/CHIPCryptoPALPSA.cpp:577]: I think we don't need to initialize this variable, as it is never used before assignment of method's output result. Similarly in other places.

File: src/crypto/CHIPCryptoPALPSA.cpp [hunk 4]
```diff
@@ -739,113 +706,113 @@ bool IsBufferContentEqualConstantTime(const void * a, const void * b, size_t n)
...
+    psa_reset_key_attributes(&attributes);
     return error;
 }
 
 CHIP_ERROR P256Keypair::Serialize(P256SerializedKeypair & output) const
 {
-    const mbedtls_ecp_keypair * keypair = to_const_keypair(&mKeypair);
-    size_t len                          = output.Length() == 0 ? output.Capacity() : output.Length();
+    const psa_ecp_keypair * keypair = to_const_keypair(&mKeypair);
+    size_t len                      = output.Length() == 0 ? output.Capacity() : output.Length();
```
Comment (kkasperczyk-no) [inline src/crypto/CHIPCryptoPALPSA.cpp:763]: Minor: could it be `const size_t len`?

Comment (Damian-Nordic) [reply src/crypto/CHIPCryptoPALPSA.cpp]: Added consts in such places.

**Pull request merged**